### DISCUSSION TITLE
Is it possible to send scheduled messages?

### DISCUSSION THREAD

**Discussion by adriangalilea** [aiogram/aiogram#1362] [Q&A] [answered]
I inspected the [send_message](https://docs.aiogram.dev/en/latest/api/methods/send_message.html) function and there's no notion of scheduling a message like you can manually do as an user.

Is this something that bot's can not do and needs to be managed manually?

Thank you.

**Comment by anorprogrammer** [comment]
Yes, you can use [apscheduler ](https://github.com/agronholm/apscheduler) to schedule a message in bot

[example](https://botfather.dev/blog/zapusk-funkczij-v-bote-po-tajmeru)

**Reply by adriangalilea** [reply]
Well, I know I can schedule it externally and manage it myself.

The question is if I can do that using the built-in method telegram has, so that I'm certain even if I stop my bot, the message will be delivered, this is crucial.

**Reply by JrooTJunior** [reply, marked as answer]
Bot API has no such kind of methods to schedule messages, so, that's mean you should use any external schedule mechanisms like apscheduler, cron, systemd timers or any other.

https://core.telegram.org/bots/api
